org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/BoundsComputer.java

```diff
@@ -46,6 +46,8 @@
 /**
  * Immutable; this class assigns each sig and field to some Kodkod relation or
  * expression, then set the bounds.
+ *
+ * @modified: Nuno Macedo, Eduardo Pessoa // [HASLab] electrum-temporal
```

Comment:
adapts the constraints the enforce the sig hierarchy to the temporal context, most relevant:
- `one` and `lone` multiplicities are enforced state-wise (but the atom may change between state)
- atoms may not change from one var prim sig to another